src/System.Linq.Expressions/src/System/Linq/Expressions/LambdaExpression.cs

```diff
@@ -80,7 +69,30 @@ ReadOnlyCollection<ParameterExpression> parameters
         /// Gets the value that indicates if the lambda expression will be compiled with
         /// tail call optimization. 
         /// </summary>
-        public bool TailCall => _tailCall;
+        public bool TailCall => TailCallCore;
+
+        protected virtual bool TailCallCore => false;
+
+        [ExcludeFromCodeCoverage] // Unreachable
+        internal virtual ReadOnlyCollection<ParameterExpression> GetOrMakeParameters()
+        {
+            throw ContractUtils.Unreachable;
+        }
+
+        [ExcludeFromCodeCoverage] // Unreachable
+        public virtual ParameterExpression GetParameter(int index)
```

Comment:
Did that now as well. There's a consideration to be made about making it public at some point so other visitors can avoid the `Parameters` allocation. I can open a separate issue for that.